### ISSUE TITLE
NameError in URL declaration of post_draft_list

### ISSUE THREAD

**Issue by depate** [DjangoGirls/tutorial-extensions#108]
Hey all!

While following the tutorial I encountered a NameError:

```bash
File "/home/patrik/dev/django-blog/blog/urls.py", line 9, in <module>
    url(r'^drafts/$', views.post_draft_list, name='post_draft_list'),
NameError: name 'url' is not defined
```
The naming convention in the beginner tutorial was `path` instead of `url`. Applying this solves the NameError.

Also the regular expression `r'^drafts/$'` is unconventional up to this point. I declared `'drafts/'` instead to follow the scheme. 

What's your opinion on the raised error?

After discussion I would volunteer to PR request a merge solving this issue.

**Comment by depate** [comment]
Ah, ok. So I can't change it in a PR, because it would break with the different django version?

Are there plans to update the `tutioral-extentions` repo?

**Comment by JonathanNguyenQuevera** [comment]
add
from django.conf.urls import url
to blog/urls.py
so:
from django.urls import path
from django.conf.urls import url
from . import views

urlpatterns = [
    path(''),
    url(r'^drafts/s', .....),
}

**Comment by depate** [comment]
Ok, I could simply change the way I reference and define URLs with this. But is it useful to so?

Why using `path` and `url` in parallel? What difference does it make?

**Comment by das-g** [comment]
> Ah, ok. So I can't change it in a PR, because it would break with the different django version?

No, please *do* file a PR. We should aim to get the extensions as up-to-date with the tutorial proper as possible.

**Comment by das-g** [comment]
> Why using `path` and `url` in parallel? What difference does it make?

Unless you need the expressive power of regular expressions for some URLs but not for others, I don't see any reason to do that. In Django 2 (which is what you're using if you recently followed the tutorial) stick to the much simpler `path` unless you have a good reason not to.

**Comment by depate** [comment]
> > Ah, ok. So I can't change it in a PR, because it would break with the different django version?
> 
> No, please _do_ file a PR. We should aim to get the extensions as up-to-date with the tutorial proper as possible.

Ok, I try to fix it then.

**Comment by depate** [comment]
> > Why using `path` and `url` in parallel? What difference does it make?
> 
> Unless you need the expressive power of regular expressions for some URLs but not for others, I don't see any reason to do that. In Django 2 (which is what you're using if you recently followed the tutorial) stick to the much simpler `path` unless you have a good reason not to.

Thanks for clarification!

**Comment by JonathanNguyenQuevera** [comment]
> Ok, I could simply change the way I reference and define URLs with this. But is it useful to so?
> 
> Why using `path` and `url` in parallel? What difference does it make?

I'm not an expert at all, In fact i'm very much so a beginner. Just wanted to share what I found to make the project work for me. There definitely could be a more elegant way to accomplish this fix.

**Comment by depate** [comment]
Thanks for showing this alternative solute as well! 

As far as I understood it, `url` is still valid for special cases in which one has to rely on regular expressions. 

Maybe a chapter or hint about this scenario in the tutorial would be nice.

**Comment by depate** [comment]
Hey,
I opened a pull request #110 to fix this.

**Comment by JonathanNguyenQuevera** [comment]
> Hey,
> I opened a pull request #110 to fix this.

Thank you!

**Comment by sarajoha** [comment]
Hi all! Is this issue closed? I saw that PR #110 solves this problem but I don't see the changes in the extension. 
PD: I'm new in git and not entirely sure of how it works.

**Comment by das-g** [comment]
> Hi all!

Hi Sara

> Is this issue closed?

No, in the issue tracker it's still open, as you can see at the top https://github.com/DjangoGirls/tutorial-extensions/issues/108:

![image](https://user-images.githubusercontent.com/97746/53155160-be0a9880-35bc-11e9-8416-7a85169e4e74.png)

> I saw that PR #110 solves this problem

Yeah, looks like it does.

Cool! #110 is already merged.

> but I don't see the changes in the extension.

Oh? On https://tutorial-extensions.djangogirls.org/en/homework/ it looks to me like the changes from https://github.com/DjangoGirls/tutorial-extensions/pull/110/files are all present. Were you maybe looking at another of the extension chapters? Not all have been adapted for Django 2, yet.

If you can confirm that this issue is indeed fixed, let's close it. If it isn't fixed, please indicate what's still wrong.

**Comment by sarajoha** [comment]
Hi Rafael!   Thanks for answering so quickly :)
I'm sorry, I wasn't specific enough, the English version is fixed but I was looking at the _spanish_ version: https://tutorial-extensions.djangogirls.org/es/homework/  
That's the one I was referring to, it still has `url`.

**Comment by das-g** [comment]
The Spanish version will be fixed with #116. If you can, review it! :smiley:

**Comment by sarajoha** [comment]
Awesome! Thanks for the info, I just reviewed it 👍
